tiledb/sm/misc/utils.cc

```diff
@@ -350,6 +350,35 @@ std::string to_str(const T& value) {
   return ss.str();
 }
 
+bool string_smaller(
+    const char* a, uint64_t size_a, const char* b, uint64_t size_b) {
+  // Check common prefix of size `size`
+  auto size = (uint64_t)std::min(size_a, size_b);
+  for (uint64_t i = 0; i < size; ++i) {
+    if (a[i] < b[i])
+      return true;
+    if (a[i] > b[i])
+      return false;
+  }
```

Comment:
`memcmp` is optimized to be faster than a simple loop (at least on Linux)
https://stackoverflow.com/questions/21106801/why-is-memcmp-so-much-faster-than-a-for-loop-check

Use this instead:
```
const int n = memcmp(a, b, size);
if (n < 0)
  return true;
if (n > 0)
  return false;
```

Reply:
I simplified for now by working only on `std::string`, so I removed that functions.